src/answers-search-bar.js

```diff
@@ -0,0 +1,519 @@
+
+    // Check if sessionsOptIn data is stored in the URL. If it is, prefer that over
+    // what is in parsedConfig.
+    const sessionOptIn = storage.get(StorageKeys.SESSIONS_OPT_IN);
+    if (!sessionOptIn) {
+      storage.set(
+        StorageKeys.SESSIONS_OPT_IN,
+        { value: parsedConfig.sessionTrackingEnabled, setDynamically: false });
+    } else {
+      // If sessionsOptIn was stored in the URL, it was stored only as a string.
+      // Parse this value and add it back to storage.
+      storage.set(
+        StorageKeys.SESSIONS_OPT_IN,
+        { value: (/^true$/i).test(sessionOptIn), setDynamically: true });
+    }
+
+    parsedConfig.noResults && storage.set(StorageKeys.NO_RESULTS_CONFIG, parsedConfig.noResults);
+    const isSuggestQueryTrigger =
+      storage.get(StorageKeys.QUERY_TRIGGER) === QueryTriggers.SUGGEST;
+    if (storage.has(StorageKeys.QUERY) && !isSuggestQueryTrigger) {
+      storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.QUERY_PARAMETER);
+    }
+
+    const context = storage.get(StorageKeys.API_CONTEXT);
+    if (context && !isValidContext(context)) {
+      storage.delete(StorageKeys.API_CONTEXT);
+      console.error(`Context parameter "${context}" is invalid, omitting from the search.`);
+    }
+
+    if (storage.get(StorageKeys.REFERRER_PAGE_URL) === undefined) {
+      storage.set(
+        StorageKeys.REFERRER_PAGE_URL,
+        urlWithoutQueryParamsAndHash(document.referrer)
+      );
+    }
+
+    this._services = getServices(parsedConfig, storage);
+
+    this._eligibleForAnalytics = parsedConfig.businessId != null;
+    // TODO(amullings): Initialize with other services
+    if (this._eligibleForAnalytics) {
+      this._analyticsReporterService = new AnalyticsReporter(
+        parsedConfig.experienceKey,
+        parsedConfig.experienceVersion,
+        parsedConfig.businessId,
+        parsedConfig.analyticsOptions,
+        parsedConfig.environment);
+
+      // listen to query id updates
+      storage.registerListener({
+        eventType: 'update',
+        storageKey: StorageKeys.QUERY_ID,
+        callback: id => this._analyticsReporterService.setQueryId(id)
+      });
+
+      this.components.setAnalyticsReporter(this._analyticsReporterService);
+      initScrollListener(this._analyticsReporterService);
+    }
+
+    this.core = new Core({
+      apiKey: parsedConfig.apiKey,
+      storage: storage,
+      experienceKey: parsedConfig.experienceKey,
+      fieldFormatters: parsedConfig.fieldFormatters,
+      experienceVersion: parsedConfig.experienceVersion,
+      locale: parsedConfig.locale,
+      analyticsReporter: this._analyticsReporterService,
+      onVerticalSearch: parsedConfig.onVerticalSearch,
+      onUniversalSearch: parsedConfig.onUniversalSearch,
+      environment: parsedConfig.environment,
+      componentManager: this.components
+    });
+
+    if (parsedConfig.onStateChange && typeof parsedConfig.onStateChange === 'function') {
```

Comment:
do we need to support onStateChange? I don't think the redirect searchbar ever touches the client's window.history